packages/contracts/src/manager/facets/ManagerFacet.sol

```diff
@@ -0,0 +1,251 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.3;
+
+import {Modifiers} from "../libraries/LibAppStorage.sol";
+import {AccessControlStorage} from "../libraries/AccessControlStorage.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "../../dollar/interfaces/IUbiquityDollarToken.sol";
+import "../../dollar/interfaces/ICurveFactory.sol";
+import "../../dollar/interfaces/IMetaPool.sol";
+import "../../dollar/core/TWAPOracleDollar3pool.sol";
+
+contract ManagerFacet is Modifiers {
+    // TODO Add a generic setter for extra addresses that needs to be linked
+    function setTwapOracleAddress(address _twapOracleAddress)
+        external
+        onlyAdmin
+    {
+        s.twapOracleAddress = _twapOracleAddress;
+        // to be removed
+
+        TWAPOracleDollar3pool oracle =
+            TWAPOracleDollar3pool(s.twapOracleAddress);
+        oracle.update();
+    }
+
+    function setDollarTokenAddress(address _dollarTokenAddress)
+        external
+        onlyAdmin
+    {
+        s.dollarTokenAddress = _dollarTokenAddress;
+    }
+
+    function setCreditTokenAddress(address _creditTokenAddress)
+        external
+        onlyAdmin
+    {
```

Comment:
all tokens can be part of the diamond and thus we can remove them from this storage

Reply:
any ERC20 / ERC1155 can be a facet of the diamond that way when targeting facets of the diamond (like the twap) inside hooks will be cheaper